### DISCUSSION TITLE
I004 isort found an unexpected blank line in imports

### DISCUSSION THREAD

**Discussion by cidrblock** [ansible/ansible-navigator#715] [Linters, linting, formatting, pre-commit, CI] [answered]
https://pypi.org/project/flake8-isort/

This is the code:

```
"""Run the :exec subcommand."""
import os
import shlex

from typing import List
from typing import Optional
from typing import Tuple
from typing import Union

from . import _actions as actions
from ..app import App
from ..configuration_subsystem import ApplicationConfiguration
from ..configuration_subsystem.definitions import Constants
from ..runner import Command
```

The blank line in the `froms` catches this error.

I like the blank line here as a delimiter between py and internal

**Comment by cidrblock** [comment]
another example of the same

```
import difflib
import json
import os

from pathlib import Path
from typing import Generator
from typing import Union

import pytest

from ..._common import fixture_path_from_request
from ..._common import update_fixtures
from ..._tmux_session import TmuxSession
from ..._interactions import SearchFor
from ..._interactions import Step
from ....defaults import FIXTURES_DIR
```
The error is thrown for the blank line between os and pathlib, the blank line separates the 2

**Reply by webknjaz** [reply]
@cidrblock I don't think you should attempt running isort, it's disabled for two reasons — it's not agreed and it's not configured. I think some of these things need to be configured so that different tools would rely on the same config. Plus that specific plugin is known to behave weirdly and I often disable it: https://github.com/ansible/pylibssh/blob/0bd71bb/.flake8#L38. Maybe it's gotten better since the last time I checked it out but I haven't had a chance to reevaluate it yet. The bottom line is that it needs to be checked together with the tool config, like https://github.com/ansible/pylibssh/blob/0bd71bb/.isort.cfg and I don't think we're ready for that yet.
We'll probably need to configure how it sees different sections/categories of imports and what the expected separator is before even starting to think about this.

**Reply by cidrblock** [reply]
This was found during a review of the vetting check, it wasn't run locally 

https://github.com/ansible/ansible-navigator/pull/624/checks?check_run_id=4493487864

Maybe I misunderstood that's what was being vetted?

**Comment by cidrblock** [comment, marked as answer]
isort has been implemented via a pre-commit hook :)
